### PR TITLE
New Page: Fresh Install - Temperature Scale step

### PR DESCRIPTION
**Pull request by arctouch-marceloaquino**
This PR intends to bring to master the following items:

- Build folder included on .gitignore file
- Some assets and icons for future usage
- Fresh Install page ( temperature scale selection step ) with its stylesheet file
- Fresh install page call on App.tsx

JIRA Items:

- Task 4
- Subtask 26

** The reason for the page call on the App.tsx is for testing purpose **

### CODE REVIEW

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+export default class FreshInstallPage extends React.Component<{}, { selectedTemperatureScale: string }> {
+
+    handleButtonClick = (selectedTemperatureScale: 'C' | 'F') => {
+        if (this.state === null || !(this.state.selectedTemperatureScale === selectedTemperatureScale)) {
+            this.setState({
+                selectedTemperatureScale
+            });
+
+            localStorage.setItem('temperatureScale', selectedTemperatureScale);
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]: Following the SOLID principles, I'd recommend us to separate all persistence logic in a separated class from the UI logic that is present in this component.

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import '../fresh_install/FreshInstallPage.css';
+import path4 from "../../images/path4.svg";
+import classNames from 'classnames';
+
+export default class FreshInstallPage extends React.Component<{}, { selectedTemperatureScale: string }> {
+
+    handleButtonClick = (selectedTemperatureScale: 'C' | 'F') => {
+        if (this.state === null || !(this.state.selectedTemperatureScale === selectedTemperatureScale)) {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]: Every state should be initialized on the constructor:

```js
class FreshInstallPage extends React.Component {
    constructor(props: Props) {
        super(props);
        this.state = { selectedTemperatureScale: null };
    }
}
```

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import '../fresh_install/FreshInstallPage.css';
+import path4 from "../../images/path4.svg";
+import classNames from 'classnames';
+
+export default class FreshInstallPage extends React.Component<{}, { selectedTemperatureScale: string }> {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]: For the code readability, I'd suggest us to extract the type definitions in `type` constants, like:

```js
type Props = { ... };
type State = { ... };

class FreshInstallPage extends React.Component<Props, State> {
```

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    public moveToLocation() {
+        console.log("Moving");
+    }
+
+    renderCelsiusButton = () => {
+        const isClicked = this.state !== null && this.state.selectedTemperatureScale === 'C';
+        const divClasses = classNames({
+            ButtonActive: isClicked,
+            ButtonCelsius: true,
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]: For those values that are always `true`, we can move the class name outside the json property, like:
```js
const divClasses = classNames('ButtonCelsius', {
    ButtonActive: isClicked,
    Button: !isClicked
});
```

For more info: https://github.com/JedWatson/classnames#usage

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            Button: !isClicked
+        });
+
+        const textClasses = classNames({
+            TextActive: isClicked,
+            Text: true,
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]: Same above

```js
const textClasses = classNames('Text', 'ButtonText', {
    TextActive: isClicked,
});
```

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    renderFahrenheit = () => {
+        const isClicked = this.state !== null && this.state.selectedTemperatureScale === 'F';
+        const divClasses = classNames({
+            ButtonActive: isClicked,
+            Button: !isClicked
+        });
+
+        const textClasses = classNames({
+            TextActive: isClicked,
+            Text: true,
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]: Same above

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            ButtonText: true
+        });
+
+        return (
+            <div className={divClasses} onClick={((e) => this.handleButtonClick('C'))}>
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]:
```suggestion
            <div className={divClasses} onClick={(e) => this.handleButtonClick('C')}>
```

File: src/pages/fresh_install/FreshInstallPage.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        });
+
+        const textClasses = classNames({
+            TextActive: isClicked,
+            Text: true,
+            ButtonText: true
+        });
+
+        return (
+            <div className={divClasses} onClick={((e) => this.handleButtonClick('F'))}>
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.tsx]:
```suggestion
            <div className={divClasses} onClick={(e) => this.handleButtonClick('F')}>
```

File: src/pages/fresh_install/FreshInstallPage.css [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+.FreshInstallPage {
+    min-height: 100vh;
+    max-width: 100%;
+    background-image: url("../../images/bg-settings-first-access@2x.png");
+    background-size: cover;
+}
+
+.Path4 {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.css]: What does `Path4` mean? Can we use another more meaningful name?

File: src/pages/fresh_install/FreshInstallPage.css [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    height: 45px;
+    object-fit: contain;
+    border-radius: 15px;
+    box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.15);
+    background-color: rgba(255, 255, 255, 0.35);
+    display: flex;
+    align-items: center;
+}
+
+.DivContent {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.css]: I don't think adding the component type on a class name is a good idea. What if we need to use it in another component that is not `<div>`? The same if we need to change the tag type to another one.

```suggestion
.Content {
```

File: src/pages/fresh_install/FreshInstallPage.css [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+.Next {
+    display: flex;
+    align-items: center;
+    height: 54px;
+    color: #ffffff;
+}
+
+.DivNext {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.css]: Same above

File: src/pages/fresh_install/FreshInstallPage.css [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+.ButtonText {
+    width: 120px;
+    height: 25px;
+    color: #2f72bb;
+    display: inline-block;
+    vertical-align: middle;
+}
+
+.TextActive {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.css]: What CSS naming convention we are using on this project? I was not able to identify any convention...

As a suggestion, I'd recommend using [BEM](http://getbem.com/naming/) that is being very useful for other projects we are working on.

File: src/pages/fresh_install/FreshInstallPage.tsx (hunk 3 above)
Comment (arctouch-marceloaquino) [reply src/pages/fresh_install/FreshInstallPage.tsx]: Great, thanks for the suggestion. Done !

File: src/pages/fresh_install/FreshInstallPage.tsx (hunk 1 above)
Comment (arctouch-marceloaquino) [reply src/pages/fresh_install/FreshInstallPage.tsx]: You are absolutely correct. I created a TemperatureScaleService inside services folder, so now every page that wants to get or set the value, could use their methods to do it.

File: src/pages/fresh_install/FreshInstallPage.css (hunk 9 above)
Comment (arctouch-marceloaquino) [reply src/pages/fresh_install/FreshInstallPage.css]: Path4 is the name of the file that represents the "arrow" near to "Next" text on the bottom of the screen. I'll rename it. Thank you.

File: src/pages/fresh_install/FreshInstallPage.tsx (hunk 2 above)
Comment (arctouch-caiodoneda) [reply src/pages/fresh_install/FreshInstallPage.tsx]: 👍
Also, the constructor method is not necessary if you want to just initialize the state.

You could do something like.

```
class FreshInstallPage extends React.Component {
        state = { selectedTemperatureScale: null };
}
```

File: src/pages/fresh_install/FreshInstallPage.css [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+.FreshInstallPage-nextArrow {
+    object-fit: contain;
+    transform: rotate(-180deg);
+    margin-left: 10px;
+}
+
+@media only screen and (max-width: 320px) {
+    .DivNext {
```
Comment (arctouch-jonathansouza) [inline src/pages/fresh_install/FreshInstallPage.css]: This class is not being used.

**Pull request merged**